docs/transpile/defaults_and_configuration_options.mdx

```diff
@@ -1,5 +1,415 @@
+  * 'as_soon_as_possible': Schedule instructions greedily, as early as possible on a qubit resource. (alias: 'asap') 
+  * 'as_late_as_possible': Schedule instructions late, that is, keeping qubits in the ground state when possible. (alias: 'alap') 
+
+**UNKNOWN**
+
+- callback (Callable[[BasePass, DAGCircuit, float, PropertySet, int], Any]) - A callback function that is called after each pass execution. The function is called with these keyword arguments:
+  * pass_: the pass being run. 
+  * dag: the dag output of the pass. 
+  * time: the time to execute the pass. 
+  * property_set: the property set. 
+  * count: the index for the pass execution. 
+  
+  <Admonition type="note"> The arguments passed expose the internals of the pass manager and are subject to change as the pass manager internals change. If you intend to reuse a callback function over multiple releases, verify that the arguments being passed are the same. 
+  </Admonition>
+  
+  To use the callback feature, define a function that will take in kwargs dict and access the variables. For example:
+
+  ```python
+  def callback_func(**kwargs):
+      pass_ = kwargs['pass_']
+      dag = kwargs['dag']
+      time = kwargs['time']
+      property_set = kwargs['property_set']
+      count = kwargs['count']
+      ...
+  transpile(circ, callback=callback_func)
+  ```
+
+- output_name (str | List[str]) -  A list with strings to identify the output circuits. The length of the list should be the same length as the circuits parameter.
+
+
+## Commonly used parameters 
+ 
+<span id="optimization-level"></span>
+### Optimization level
+
+Decomposing quantum circuits into the basis gate set of the target device and the addition of SWAP gates needed to match hardware topology causes an increase the depth and gate count of quantum circuits. To mitigate this increased complexity, you can set the `optimization_level`.  Setting this value calls an optimization routine that optimizes the transpilation process by combining or eliminating gates and by optionally using algorithms to find an optimal layout (depending on the level chosen). 
+
+In some cases these methods are so effective the output circuits have lower depth than the inputs. In other cases, not much can be done, and the computation may be difficult to perform on noisy devices. Choosing the best optimization level might take trial and error, as it depends heavily on the circuit being transpiled and the backend being targeted. By default, `optimization_level=1` is used.
+
+The `optimization_level` setting specifies how much optimization to perform on the circuits. Higher levels generate more optimized circuits by increasing the heuristic effort spent trying to optimize, at the expense of longer compile times.  The default level is 1. 
```

Comment:
should this sentence be removed? it looks like it just duplicates what is said in the intro